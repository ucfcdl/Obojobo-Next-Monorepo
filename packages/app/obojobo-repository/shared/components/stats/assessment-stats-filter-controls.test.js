import React from 'react'
import renderer from 'react-test-renderer'
import AssessmentStatsFilterControls from './assessment-stats-filter-controls'

describe('AssessmentStatsFilterControls', () => {
	test('AssessmentStatsFilterControls renders correctly', () => {})

	test.each`
		showIncompleteAttempts | showPreviewAttempts | showAdvancedFields
		${false}               | ${false}            | ${false}
		${false}               | ${false}            | ${true}
		${false}               | ${true}             | ${false}
		${false}               | ${true}             | ${true}
		${true}                | ${false}            | ${false}
		${true}                | ${false}            | ${true}
		${true}                | ${true}             | ${false}
		${true}                | ${true}             | ${true}
	`(
		'<AssessmentStatsFilterControls showIncompleteAttempts=$showIncompleteAttempts showPreviewAttempts=$showPreviewAttempts showAdvancedFields=$showAdvancedFields /> matches snapshot',
		({ showIncompleteAttempts, showPreviewAttempts, showAdvancedFields }) => {
			const component = renderer.create(
				<AssessmentStatsFilterControls
					onChangeFilterSettings={jest.fn()}
					filterSettings={{ showIncompleteAttempts, showPreviewAttempts, showAdvancedFields }}
				/>
			)
			const tree = component.toJSON()
			expect(tree).toMatchSnapshot()
		}
	)

	test('Inputs work as expected', () => {
		const onChangeFilterSettings = jest.fn()

		const component = renderer.create(
			<AssessmentStatsFilterControls
				onChangeFilterSettings={onChangeFilterSettings}
				filterSettings={{
					showIncompleteAttempts: false,
					showPreviewAttempts: false,
					showAdvancedFields: false
				}}
			/>
		)

		const checkboxShowIncomplete = component.root.findByProps({
			className: 'show-incomplete-attempts'
		})
		const checkboxShowPreview = component.root.findByProps({ className: 'show-preview-attempts' })
		const checkboxAdvanced = component.root.findByProps({ className: 'show-advanced-fields' })

		expect(onChangeFilterSettings).not.toHaveBeenCalled()

		// Click on checkboxShowIncomplete
		checkboxShowIncomplete.props.onChange({ target: { checked: true } })
		expect(onChangeFilterSettings).toHaveBeenCalledTimes(1)
		expect(onChangeFilterSettings).toHaveBeenCalledWith({
			showIncompleteAttempts: true,
			showPreviewAttempts: false,
			showAdvancedFields: false
		})

		// Click on checkboxShowPreview
		checkboxShowPreview.props.onChange({ target: { checked: true } })
		expect(onChangeFilterSettings).toHaveBeenCalledTimes(2)
		expect(onChangeFilterSettings).toHaveBeenCalledWith({
			showIncompleteAttempts: false,
			showPreviewAttempts: true,
			showAdvancedFields: false
		})

		// Click on showAdvancedFields
		checkboxAdvanced.props.onChange({ target: { checked: true } })
		expect(onChangeFilterSettings).toHaveBeenCalledTimes(3)
		expect(onChangeFilterSettings).toHaveBeenCalledWith({
			showIncompleteAttempts: false,
			showPreviewAttempts: false,
			showAdvancedFields: true
		})
	})
})
