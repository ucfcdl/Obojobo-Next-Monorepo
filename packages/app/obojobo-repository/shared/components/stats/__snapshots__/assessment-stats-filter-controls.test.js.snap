// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=false showPreviewAttempts=false showAdvancedFields=false /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={false}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={false}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=false showPreviewAttempts=false showAdvancedFields=true /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={false}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={true}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=false showPreviewAttempts=true showAdvancedFields=false /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={false}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={true}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={false}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=false showPreviewAttempts=true showAdvancedFields=true /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={false}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={true}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={true}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=true showPreviewAttempts=false showAdvancedFields=false /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={true}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={false}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=true showPreviewAttempts=false showAdvancedFields=true /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={true}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={false}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={true}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=true showPreviewAttempts=true showAdvancedFields=false /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={true}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={true}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={false}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;

exports[`AssessmentStatsFilterControls <AssessmentStatsFilterControls showIncompleteAttempts=true showPreviewAttempts=true showAdvancedFields=true /> matches snapshot 1`] = `
<div
  className="repository--assessment-stats-filter-controls"
>
  <div
    className="container"
  >
    <label>
      <input
        checked={true}
        className="show-incomplete-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include incomplete attempt data
      </span>
    </label>
    <label>
      <input
        checked={true}
        className="show-preview-attempts"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include preview attempt data
      </span>
    </label>
    <hr />
    <label>
      <input
        checked={true}
        className="show-advanced-fields"
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        Include advanced fields
      </span>
    </label>
  </div>
</div>
`;
